### ISSUE TITLE
Organizing directories

### ISSUE THREAD

**Issue by natario1** [parse-community/Parse-SDK-Android#679]
I’d like to address #98 , but before doing that, I’d like to discuss file/cache storage as it currently is.
Most of all I would like to get rid of the deprecated `Parse.getParseDir()` which is still used here and there. There are:

- `Parse.getParseDir()` annotated as deprecated, and I agree
- `Parse.getParseCacheDir()`
- `Parse.getParseFilesDir()`

Here’s where we use them:

- `getParseDir() + "/LocalId"`: used by LocalIdManager
- `getParseDir() + "/CommandCache"`: used by `ParseCommandCache`
- `getParseDir() + "/currentUser"`: used by `ParseCorePlugins`
- `getParseDir() + "/currentInstallation"`: used by `ParseCorePlugins`
- `getParseDir() + "/currentConfig"`: used by `ParseCorePlugins`
- `getParseCacheDir() + "/applicationId"`: used by `Parse`
- `getParseCacheDir() + "/GCMRegistrar"`: used by `GcmRegistrar`
- `getParseCacheDir() + "/files"` : used by ParseFileController for `ParseFile` temporary and cache files
- `getParseFilesDir() + "/push"` : used by `PushRouter`

So we should move all `getParseDir()` (deprecated) to either `getParseCacheDir()` or `getParseFilesDir()`. But looking at this, I see no pattern. What should go to `cacheDir`, and what to `filesDir`?

I know the implementation won't be easy to avoid breaking current caches, so we will have to support legacy dirs for a while etc. But here I would just like to understand the intended usage of `cacheDir` and `filesDir`. We actually store no files, these are all just caches of something. Even `ParseFile` are stored in `cacheDir/files`...

**In short** 
We need a new home for LocalId, CommandCache, currentUser, currentInstallation and currentConfig. Should it be `cacheDir` or `filesDir` ? And actually, couldn't we use `cacheDir` for everything?

**Comment by natario1** [comment]
Suggestions:

- LocalId: move to `cacheDir`
- CommandCache: move to `cacheDir`
- currentUser: move to `filesDir`
- currentInstallation: move to `filesDir`
- currentConfig: move to `filesDir`
- applicationId: I don’t think this belongs to cacheDir, move to `filesDir`

and leave the others where they are. If I got this correctly this means that when the user clears app cache, `currentUser`, `currentInstallation`, `currentConfig`, `applicationId` and push history are kept. The rest is cleared.
What do you think?

**Comment by ZeekHuge** [comment]
Hi, do not intend to hijack this thread. But I think my question is very much related.
I want to keep files locally, for a long duration of time. Also, I wish to keep a lot of other object in the local data store using *.pinInBackground() or similar.

So my question is : where will all the `pinned` objects get saved ? and can I change something in config so that parse saves the downloaded Files (using ParseFile.getFile()) in a different dir ?
also, is `pinning` good for heavy data lifting ?
